render/gles2/renderer.c

```diff
@@ -243,9 +243,9 @@ static int gles2_get_dmabuf_modifiers(struct wlr_renderer *wlr_renderer,
 }
 
 static bool gles2_check_import_dmabuf(struct wlr_renderer *wlr_renderer,
-		struct wlr_dmabuf_buffer *dmabuf) {
+		struct wlr_dmabuf_attributes *attribs) {
 	struct wlr_gles2_renderer *renderer = gles2_get_renderer(wlr_renderer);
```

Comment:
We might want to do sanity checks on the buffer as well (e.g. for YUV) at some point that's why I passed in the buffer and not only the attribs but since it's not a public function we can change that back later.

Reply:
I'm starting to think it would be better to remove this function and try to import+destroy the DMA-BUF in the protocol implementation instead. Thoughts?

Reply:
You mean rather do it in wlr_linux_dmabuf? I'd rather leave it where it is since YUV will need more stuff in egl etc (I should really sit down and do the YUV stuff since some things will become clearer then, but I'm currently a bit short on time due to RL stuff)

Reply:
Okay, let's keep it there for now. I'm curious though:

1. "We might want to do sanity checks on the buffer as well" → what piece of information relevant to GL is there in `wlr_dmabuf_buffer` and not in `wlr_dmabuf_attributes`?
2. Can't we do these checks in `wlr_gles2_texture_from_dmabuf`?

Reply:
1.)  Things like say NV12 having two planes can be done with attributes only but e.g. checking fd seekability can not
2.) From my understanding of the protocol you want to do these things _before_ you finally switch the resource from params_resource to buffer_resoure so doing it in wlr_gles2_texture_from_dmabuf would be too late

Reply:
>checking fd seekability can not

Hum, why that? Can't you just `fseek(attribs->fd[i])` in the renderer code?

>From my understanding of the protocol you want to do these things before you finally switch the resource from params_resource to buffer_resoure so doing it in wlr_gles2_texture_from_dmabuf would be too late

Well, the idea would be to call `wlr_gles2_texture_from_dmabuf` before switching the resource to a buffer, and destroy the texture immediately. `gles2_check_import_dmabuf` already does this with `wlr_egl`.

Reply:
> Hum, why that? Can't you just fseek(attribs->fd[i]) in the renderer code?
That would work too. I'm fine with that either way.

> Well, the idea would be to call wlr_gles2_texture_from_dmabuf before switching the resource to a buffer, and destroy > the texture immediately. gles2_check_import_dmabuf already does this with wlr_egl.
I guess I'm confused now. So the idea would be to replace the call of ```wlr_renderer_check_import_dmabuf``` to ```wlr_gles2_texture_from_dmabuf``` + destroy to get rid of that function? Fine with me as well although having a explicit _check_ function makes the code more self documenting (at least for me).

Reply:
>Fine with me as well although having a explicit check function makes the code more self documenting (at least for me).

Yeah, the idea is just to reduce the `wlr_renderer` API surface.